src/client/debugger/jupyter/debugger.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+import { debug, NotebookDocument, DebugSession, DebugSessionOptions, DebugConfiguration } from 'vscode';
+
+export class Debugger {
+    private resolveFunc?: (value: DebugSession) => void;
+    private rejectFunc?: (reason?: Error) => void;
+
+    readonly session: Promise<DebugSession>;
+
+    constructor(
+        public readonly document: NotebookDocument,
+        public readonly config: DebugConfiguration,
+        options?: DebugSessionOptions
+    ) {
+        this.session = new Promise<DebugSession>((resolve, reject) => {
+            this.resolveFunc = resolve;
+            this.rejectFunc = reject;
+
+            debug.startDebugging(undefined, config, options).then(undefined, reject);
+        });
+    }
+
+    resolve(session: DebugSession) {
+        if (this.resolveFunc) {
+            this.resolveFunc(session);
+        }
+    }
+
+    reject(reason: Error) {
+        if (this.rejectFunc) {
+            this.rejectFunc(reason);
+        }
+    }
+
+    async stop() {
+        void debug.stopDebugging(await this.session);
```

Comment:
Why is this async if we're ignoring the promise returned by `debug.stopDebuggin`

Reply:
because we have to wait to get the session